soroban-env-host/src/native_contract/token/admin.rs

```diff
@@ -2,18 +2,23 @@ use crate::host::Host;
 use crate::native_contract::base_types::Address;
 use crate::native_contract::token::storage_types::DataKey;
 use crate::HostError;
-use soroban_env_common::{Env, TryIntoVal};
+use soroban_env_common::{Env, StorageType, TryIntoVal};
 
 // Metering: covered by components
 pub fn read_administrator(e: &Host) -> Result<Address, HostError> {
     let key = DataKey::Admin;
-    let rv = e.get_contract_data(key.try_into_val(e)?)?;
+    let rv = e.get_contract_data(key.try_into_val(e)?, StorageType::MERGEABLE)?;
     rv.try_into_val(e)
 }
 
 // Metering: covered by components
 pub fn write_administrator(e: &Host, id: Address) -> Result<(), HostError> {
     let key = DataKey::Admin;
-    e.put_contract_data(key.try_into_val(e)?, id.try_into_val(e)?)?;
+    e.put_contract_data(
+        key.try_into_val(e)?,
+        id.try_into_val(e)?,
+        StorageType::MERGEABLE,
```

Comment:
So the admin can only be created by calling `set_admin` (which won't work if an admin doesn't already exist), or by calling `init_asset`, which can only be called once because there's an `asset_info` check and it's unique. This is a little confusing though. What do you think?

Reply:
I see. It's not immediately obvious, but probably fine as is. Going forward we might just use contract instance storage for that and make things more clear (I'm really tempted by this proposal).